lib/stacks/iam.ts

```diff
@@ -422,7 +484,11 @@ export class IAMStack extends Stack {
             this.mlSpaceNotebookRole = new Role(this, 'mlspace-notebook-role', {
                 roleName: mlSpaceNotebookRoleName,
                 assumedBy: notebookPolicyAllowPrinciples,
-                managedPolicies: [notebookPolicy],
+                managedPolicies: [
+                    notebookPolicy,
+                    this.mlspaceEndpointConfigInstanceConstraintPolicy,
```

Comment:
Also should only be added if `MANAGE_IAM_POLICIES` is set

Reply:
I'll wait to see if you agree with my comment above. I think if we're creating the notebook role these are being applied regardless of if they're predefined or we're creating them during cdk deploy.

Reply:
These do need to be conditionally added

Reply:
Yep based on what we talked about I made those changes along with responding to your other comments